plugins/catalog-backend/src/service/createRouter.ts

```diff
@@ -228,6 +231,28 @@ export async function createRouter(
     });
   }
 
+  if (orchestrator) {
+    router.post('/validate-entity', async (req, res) => {
+      const bodySchema = z.object({
+        entity: z.unknown(),
+        location: z.string(),
+      });
+
+      const body = await validateRequestBody(req, bodySchema);
+
+      const entity = setRequiredEntityLocationAnnotations(
+        body.entity as Entity,
+        body.location,
+      );
+      const processingResult = await orchestrator.process({
+        entity,
+      });
+      if (!processingResult.ok)
+        throw new InputError(processingResult.errors[0].message);
```

Comment:
It would be nice to get a more comprehensive error output with all errors in it - if you look into the `@backstage/errors` package it has a `serializeError` function. So it would be nice if we could make the response here

```ts
return res.status(400).json({
  errors: processingResult.errors.map(e => serializeError(e)),
});
```